### PR TITLE
[feature/pubsub-batching] Incorporate Review comments for Bulk Subscribe

### PR DESCRIPTION
**Pull request by DeepanshuA**
Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

# Description

This PR intends to incorporate review comments for Bulk Subscribe part, provided under https://github.com/dapr/dapr/pull/5256:

## Issue reference

Address review comments for Bulk Subscribe part, provided with: #5256

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/channel/testing/channel_mock.go [hunk 1]
```diff
@@ -82,9 +84,11 @@ func (_m *MockAppChannel) Init() {
```
Comment (ItalyPaleAle) [inline pkg/channel/testing/channel_mock.go:84]: (Can't leave a commend on L82) Doesn't the `Init` method need a lock too?

File: pkg/channel/testing/channel_mock.go [hunk 2]
```diff
@@ -82,9 +84,11 @@ func (_m *MockAppChannel) Init() {
...
 // InvokeMethod provides a mock function with given fields: ctx, req
 func (_m *MockAppChannel) InvokeMethod(ctx context.Context, req *v1.InvokeMethodRequest) (*v1.InvokeMethodResponse, error) {
+	_m.mutex.Lock()
 	if _m.requestsReceived != nil {
 		_m.requestsReceived[req.Message().Method] = req
 	}
+	_m.mutex.Unlock()
 	ret := _m.Called(ctx, req)
 
 	var r0 *v1.InvokeMethodResponse
```
Comment (ItalyPaleAle) [inline pkg/channel/testing/channel_mock.go:94]: (For line 114) shoulodn't there be a lock on `GetInvokeRequest` too?

**Review by ItalyPaleAle — APPROVED** [review]
Approved but think you need a couple more locks

File: pkg/channel/testing/channel_mock.go (hunk 1 above)
Comment (DeepanshuA) [reply pkg/channel/testing/channel_mock.go:86]: Agreed, incorporated.

File: pkg/channel/testing/channel_mock.go (hunk 2 above)
Comment (DeepanshuA) [reply pkg/channel/testing/channel_mock.go:96]: Agreed, incorporated.

**Pull request merged**